### PR TITLE
Melee weapons rebalance [WIP]

### PR DESCRIPTION
**Pull request by Leander-0**
## About the PR 

This is a very much work in progress to rebalance most if not all the melee weapons to accommodate a more stamina focused combat, also some weapons are gonna use a new damage modifier called Bloodletting (Also WIP) that will drain blood directly upon hit.

Everything here is subject to change and to feedback, so this might not be the final damage values.

Any overlooked weapons that we could have forgotten it would be nice to point it out thanks :^)  

### Other changes
 
The janitor mops and the folding chair will be able to wielded but they lack a sprite when used in two hands, so anyone that volunteers in that artistic area it would be very much appreciated.

- [ ] Regular janitor mop
- [ ] Advanced mop
- [ ] Folding chair
- [ ] Shovel

**Changelog**

:cl:
- tweak: rebalance of most of the melee weapons

### CODE REVIEW

**Comment by Elijahrane** [comment]
poggers

**Comment by Vordenburg** [comment]
This will eventually depend on #730.

**Comment by Leander-0** [comment]
### Other changes besides melee weapons

bleeding coefficients for the slash and piercing damage where nerfed, this will affect all weapons and guns too.

- pierce 2 ---> 1.5
- slash 2.5 ---> 2

 Slowdown threshold when damaged slightly buffed.

- 60 damage: 0.7 ---> 0.8
- 80 damage: 0.5 ---> 0.6

**Comment by Leander-0** [comment]
We are only lacking the sprites for the 2 handed versions of the:

- Folding chair
- Mop
- Advanced mop
- Shovel

but besides that this should be ready for review.

**Comment by Vordenburg** [comment]
The `swing` field for `MeleeStaminaCost` has a bug which I'll be investigating a fix for tomorrow. It detects hits just fine but miss events are not raised.

**Comment by Leander-0** [comment]
One important change that has been done is the **Decay** value in the stamina component which is the rate of stamina heal, right now its at 

- 3 stamina per second

 but since we are going forwards a more stamina oriented combat it was buffed to 

- 5 stamina per second

the component also has a **DecayCooldown** from when one starts healing the stamina damage, right now it takes 5 seconds for stamina to start regenerating, depending on how good or bad it feels to play at this cooldown it may be tweaked.

**Comment by Vordenburg** [comment]
#752 addresses all currently known issues with the new melee components.

**Comment by Leander-0** [comment]
### more stuff changed:

- Oni stamina damage modifier was nerfed in consideration of the extra stamina damage weapons do.

- Weapons when wielded will cost less stamina to swing and hit people with.

@Elijahrane this is ready since no more changes will be applied

**Review by Elijahrane — APPROVED** [review]

File: Resources/Prototypes/Entities/Objects/Specific/Janitorial/janitor.yml [hunk 1]
```diff
@@ -11,6 +11,18 @@
...
+      types:
+        Blunt: 2
+  - type: StaminaDamageOnHit
+    damage: 5
+  - type: Wieldable
+  - type: IncreaseDamageOnWield
+    damage:
+      types:
+        Blunt: 5
+    bluntStaminaDamageFactor: 1.60 # 8 stamina damage
```
Comment (Vordenburg) [inline Resources/Prototypes/Entities/Objects/Specific/Janitorial/janitor.yml:25]: This `bluntStaminaDamageFactor` field is being set on IncreaseDamageOnWield, when it should be set on MeleeWeapon. IncreaseDamageOnWield has no such field.

Comment (Leander-0) [reply Resources/Prototypes/Entities/Objects/Specific/Janitorial/janitor.yml:25]: oh whoops i didn't notice, will fix it later

**Pull request merged**